File: Nian_Karl/QL/src/org/uva/ql/ast/type/Type.java
package org.uva.ql.ast.type;

import org.uva.ql.ast.Node;
import org.uva.ql.visitor.TypeVisitable;

public abstract class Type implements Node, TypeVisitable {

	public boolean isInt() {
		return false;
	}

	public boolean isBool() {
		return false;
	}

	public boolean isStr() {
		return false;
	}

	public abstract boolean isEqual(Type type);

	public boolean isUndefined() {
		return false;
	}
}

File: FelipezConde/QuestionnaireLanguage/AST/Types/Type.cs
using AST.Types.Visitors;

namespace AST.Types
{
    public abstract class Type : ITypeVisitable
    {
        public virtual bool IsInt()
        {
            return false;
        }

        public virtual bool IsBool()
        {
            return false;
        }

        public virtual bool IsString()
        {
            return false;
        }

        public virtual bool IsUndefined()
        {
            return false;
        }

        public abstract bool IsEqual(Type type);

        public abstract T Accept<T>(ITypeVisitor<T> visitor);

        public virtual Types.Type RetrieveType()
        {
            return new Types.UndefinedType();
        }
    }
}
